fixes #9027 feat(nimbus): hide enrollment duration for rollouts

tj-26 opened the pull request:
Because

- Because the ability to pause rollouts was disabled so, the enrollment-duration field has no meaning and it should not be visible for rollouts.

This commit

- Hides the enrolment duration if the experiment is a rollout.

<img width="1050" alt="Screenshot 2023-10-28 at 3 37 12 PM" src="https://github.com/mozilla/experimenter/assets/48300889/cad667b7-dd94-43f9-81e3-c3b79133f8f5">

fixes #9027

brennie commented on experimenter/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.test.tsx:
```diff
@@ -2076,6 +2076,44 @@ it("enables save buttons when not archived", async () => {
+  );
+
+  await waitFor(() => {
+    expect(screen.queryByTestId("proposedEnrollment")).toBeInTheDocument();
```
```suggestion
    screen.getByTestId("proposedEnrollment");
```

`getByTestId` will throw if the element is not found.

brennie commented on experimenter/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.test.tsx:
```diff
@@ -2076,6 +2076,44 @@ it("enables save buttons when not archived", async () => {
+  );
+
+  await waitFor(() => {
+    expect(screen.queryByTestId("proposedEnrollment")).not.toBeInTheDocument();
```
```suggestion
    expect(screen.queryByTestId("proposedEnrollment")).toBeNull();
```

queryByTestId returns null if not found, so we're testing if `null` will be in the document. It will never be true, but semantically it makes more sense to check for null in the return value.

eliserichards commented on experimenter/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.tsx:
```diff
@@ -801,6 +773,38 @@ export const FormAudience = ({
             </InputGroup>
           </Form.Group>
+
+          {experiment.isRollout ? (
```
This change also switches the order of "Experiment duration" and "Enrollment period", right? Can we keep the order the same (e.g. for non-rollouts, "Enrollment period" is on the left and "Experiment duration" is on the right):
<img width="2057" alt="image" src="https://github.com/mozilla/experimenter/assets/43795363/52d0546b-cc2b-4d36-acd3-b039f3954c65">

eliserichards replied:
(For context, if we change the order of these fields it will definitely confuse our users and they may enter incorrect values)

eliserichards replied:
I think we would actually like to have the field greyed out instead of totally hidden (see comment [here](https://github.com/mozilla/experimenter/issues/9027#issuecomment-1789616457)). Sorry to change the requirements on you mid-PR 😅

tj-26 replied:
Hi @eliserichards. No Worries. Thanks for letting me know. I'll make the change.

eliserichards reviewed (changes requested):
Thanks for the contribution @tj-26! I have one suggestion:

tj-26 commented:
Updated Behaviour: Now on rollouts the enrolment period field will be greyed out(disabled) instead of hidden completely.
Also, because the disabled field visibility is not great I added a tooltip that only shows on rollouts to provide more information.
<img width="611" alt="Screenshot 2023-11-06 at 10 54 44 PM" src="https://github.com/mozilla/experimenter/assets/48300889/099754eb-447c-41d4-a221-a7a4d4015656">

yashikakhurana reviewed (approved):
Thank you @tj-26 for contribution,  looks perfect 🎉 🎉
